src/L.Routing.Mapzen.js

```diff
@@ -160,50 +163,53 @@
 
     },
 
+    //creates section of the polyline based on change of travel mode for multimodal
     _getSubRoutes: function(legs) {
 
       var subRoute = [];
 
       for (var i = 0; i < legs.length; i++) {
 
-        var coord = polyline.decode(legs[i].shape, 6);
+        var coords = polyline.decode(legs[i].shape, 6);
 
-        var lastTravelType;
-        var travelTypeChangingIncides = [];
+        var transitIndices = [];
         for(var j = 0; j < legs[i].maneuvers.length; j++){
 
           var res = legs[i].maneuvers[j];
           var travelType = res.travel_type;
 
-          if(travelType !== lastTravelType && lastTravelType !== 'undefined' && travelType !=='undefined') {
-            if(res.begin_shape_index > 0) travelTypeChangingIncides.push(res.begin_shape_index);
-            if(res.transit_info) subRoute.push({ travle_type: travelType, styles: this._getTransitColor(res.transit_info.color) })
-            else subRoute.push({travle_type: travelType})
-          }
-          lastTravelType = travelType;
+          //transit_info only exists in the transit maneuvers
```

Comment:
this populates all maneuvers as subroutes, wouldn't it be more useful to divide routes into suboutes only when `travle_type` is changed?